__mocks__/dataLayer.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * This is meant to be a mock for interacting with the dataLayer.
+ */
+
+// used for GOOGLE OPTIMIZE implementation
+export const OPTIMIZE_ID = "123456";
+export const VARIANT = "1";
+
+let dataLayer = {};
```

Comment:
If we're using `dataLayer` in the Google sense, should this be an array?   Then the push is inherently defined.  Can we then do [spyOn](https://jestjs.io/docs/jest-object#jestspyonobject-methodname)?  I dunno, I _think_ it's applicable, but throwing it out there in case it allows us to have a mock that we don't have to implement.

Reply:
I do need to override the function, as I need the specific push behavior that google optimize implements.  (And hoping this can be the basis for other people to add behaviors if needed). If spyOn still makes sense, I can definitely do that.  And either way, you are right, this should be an array.

Reply:
Ah, I see there's side effects _if_ the push is an object that has a `callback` function within it.  I wonder if we could generalize that or if that's a standard patter in Google's datalayer.  Should we possibly look for any event name ending in `.callback` and run a corresponding function.

Reply:
Yeah I wasn't sure how far we wanted to go into that.  I didn't research the various dataLayer.push behavior.  I just knew for google optimize I needed this exact behavior.  I'm guessing there will be others, and people can add that in as their tests need?  Just let me know if you think spyOn would still help?

Reply:
I think your specialized implementation to support the callback is the way to go.  If we can have the dataLayer mock mimic that behavior generally (without over-engineering), that'd be rad.

Reply:
The hard part is knowing what the parameters need to be for the callback.  In this case, we call the callback function and pass in variant and optimizeID.  I don't know what the other ones are.

Reply:
After reviewing the full context, agreed.  We can always revisit.